tools/cli/src/main.rs

```diff
@@ -44,18 +52,86 @@ async fn main() -> Result<()> {
                     let _ = task_queue.add_task(body, description, timeout).await?;
                 }
                 TaskQueueCommand::ViewQueue => {
-                    let mut buffer = Vec::new();
-
-                    let _ = task_queue
-                        .querier
-                        .task_queue_view()
-                        .await?
-                        .report(&mut buffer);
-
-                    tracing::info!("{}", String::from_utf8(buffer)?);
+                    let res = task_queue.querier.task_queue_view().await?;
+                    tracing::info!("Task Queue Configuration");
+                    res.report(|line| {
+                        println!("{}", line);
+                    })?;
                 }
             }
         }
+        Command::Faucet(faucet_args) => match faucet_args.command {
+            FaucetCommand::Tap { to, amount, denom } => {
+                let to = match to {
+                    Some(to) => ctx.chain_config()?.parse_address(&to)?,
+                    None => ctx.any_client().await?.as_signing().addr.clone(),
+                };
+
+                let amount = amount.unwrap_or(FaucetCommand::DEFAULT_TAP_AMOUNT);
+                tap_faucet(&ctx, to, amount, denom).await?;
+            }
+        },
+        Command::Wallet(wallet_args) => {
+            let mut rng_lock = ctx.rng.lock().await;
+            wallet_args
+                .command
+                .run(ctx.any_client().await?, &mut *rng_lock, |line| match line {
```

Comment:
Interesting way to pass in closure to output logging.

what about returning one of those values (or a vec of them if multiple) and then doing the match normally rather than a closure.

Seems simpler, but not a blocker. Just style comment

Reply:
the reason is that the function as a whole might fail, but you might want to see logs live as it's happening before fail.

not saying it's a _good_ reason, returning a `Vec<Result<...>>` might be totally fine. good to think about